### ISSUE TITLE
Cache songs in SQLite based on user choice

### ISSUE THREAD

**Issue by arunsudhir** [arunsudhir/RadioMalayalam#6]

**Comment by ViteFalcon** [comment]
Instead of MRU, we could allow the users to choose what they want offline/cached.

**Comment by arunsudhir** [comment]
yep we talked about this.. and I forgot.. we will expose a download button in UI.. should we call it download r something like SmartStream (marketing gimmick)
